aeneas/src/x86/X86CodeGen.v3

```diff
@@ -168,6 +169,79 @@ class X86CodeGen extends MachCodeGen {
 			IntConvert =>		genIntConvert(i, v);
 			BoolNot =>		gen_op2_i("xor", X86Assembler.xor, v, i.input0(), Int.box(1));
+			IntCastF(isDouble) => {
+				var name: string, m: (X86MacroAssembler, X86Reg, SSERm) -> void;
+				var signed = isSigned(i, true);
+				if (isDouble) {
+					name = if(signed, "cvt_i2d", "cvt_ui2d");
+					m = if(signed, X86Assembler.cvtsd2si, X86MacroAssembler.cvt_d2ui);
+				} else {
+					name = if(signed, "cvt_i2f", "cvt_ui2f");
+					m = if(signed, X86Assembler.cvtss2si, X86MacroAssembler.cvt_f2ui);
+				}
+				gen_sse_f2i(name, m, v, i.input0());
+			}
+			IntViewF(isDouble) => {
+				if (isDouble) gen_sse_view_int("int_viewd", X86Assembler.movd_rm_s, v, i.input0()); // TODO: fix viewd
+				else gen_sse_view_int("int_viewf", X86Assembler.movd_rm_s, v, i.input0());
+			}
+			IntTruncF(isDouble) => {
+				var name: string, m: (X86MacroAssembler, X86Reg, SSERm) -> void;
+				var signed = isSigned(i, true);
+				if (isDouble) {
+					name = if(signed, "int_truncd", "uint_truncd");
+					m = if(signed, X86MacroAssembler.int_truncd, X86MacroAssembler.uint_truncd);
+				} else {
+					name = if(signed, "int_truncf", "uint_truncf");
+					m = if(signed, X86MacroAssembler.int_truncf, X86MacroAssembler.uint_truncf);
+				}
+				gen_sse_trunc(name, m, v, i.input0());
+			}
+			FloatPromoteF => 	gen_sse_f2d("float_promote", X86Assembler.cvtss2sd, v, i.input0());
+			FloatViewI(isDouble) => {
+				if (isDouble) gen_sse_view_float("double_viewi", X86Assembler.movd_s_rm, v, i.input0()); // TODO: fix viewi
+				else gen_sse_view_float("float_viewi", X86Assembler.movd_s_rm, v, i.input0());
+			}
+			FloatPromoteI(isDouble) => 	genFloatRoundI(i, v, isDouble);
+			FloatRoundI(isDouble) => 	genFloatRoundI(i, v, isDouble);
+			FloatRound(isDouble) => {
+				if (isDouble) gen_sse_round("roundsd", X86Assembler.roundsd, v, i.input0(), RoundingMode.TO_NEAREST);
+				else gen_sse_round("roundss", X86Assembler.roundss, v, i.input0(), RoundingMode.TO_NEAREST);
+			}
+			FloatAdd(isDouble) => {
+				if (isDouble) gen_sse_op2("addsd", X86Assembler.addsd, v, i.input0(), i.input1());
+				else gen_sse_op2("addss", X86Assembler.addss, v, i.input0(), i.input1());
+			}
+			FloatSub(isDouble) => {
+				if (isDouble) gen_sse_op2("subsd", X86Assembler.subsd, v, i.input0(), i.input1());
+				else gen_sse_op2("subss", X86Assembler.subss, v, i.input0(), i.input1());
+			}
+			FloatMul(isDouble) => {
+				if (isDouble) gen_sse_op2("mulsd", X86Assembler.mulsd, v, i.input0(), i.input1());
+				else gen_sse_op2("mulss", X86Assembler.mulss, v, i.input0(), i.input1());
+			}
+			FloatDiv(isDouble) => {
+				if (isDouble) gen_sse_op2("divsd", X86Assembler.divsd, v, i.input0(), i.input1());
+				else gen_sse_op2("divss", X86Assembler.divss, v, i.input0(), i.input1());
+			}
+			FloatBitEq(isDouble) => gen_sse_bit_eq(v, i.input0(), i.input1(), isDouble);
+			FloatEq(isDouble) => gen_sse_cmp(sseCond(i.op.opcode), v, i.input0(), i.input1(), isDouble);
```

Comment:
I think you know the condition from the operator here, so no need to call sseCond, right?

Reply:
Yep. This was before the `isDouble` patch, so I had them fall through before. I'll change that now.